lib/std/testing.zig

```diff
@@ -210,18 +210,12 @@ test "expectEqual.union(enum)" {
 
 /// This function is intended to be used only in tests. When the formatted result of the template
 /// and its arguments does not equal the expected text, it prints diagnostics to stderr to show how
-/// they are not equal, then returns an error.
+/// they are not equal, then returns an error. It depends on `expectEqualStrings()` for printing
+/// diagnostics.
 pub fn expectFmt(expected: []const u8, comptime template: []const u8, args: anytype) !void {
-    const result = try std.fmt.allocPrint(allocator, template, args);
-    defer allocator.free(result);
-    if (std.mem.eql(u8, result, expected)) return;
-
-    print("\n====== expected this output: =========\n", .{});
-    print("{s}", .{expected});
-    print("\n======== instead found this: =========\n", .{});
-    print("{s}", .{result});
-    print("\n======================================\n", .{});
-    return error.TestExpectedFmt;
+    const actual = try std.fmt.allocPrint(allocator, template, args);
+    defer allocator.free(actual);
+    return expectEqualStrings(expected, actual) catch error.TestExpectedFmt;
```

Comment:
```suggestion
    return expectEqualStrings(expected, actual);
```

why not?

Reply:
No particular reason for keeping `error.TextExpectedFmt`.  Done.

Reply:
I made the change locally and force pushed rather than using the 'Commit suggestion' button.  Not sure if what would have been best.  Just trying to avoid github adding merge commits or otherwise messing things up.